chain/indexer/src/streamer/mod.rs

```diff
@@ -51,22 +51,32 @@ async fn build_streamer_message(
     let chunks = fetch_chunks(&client, chunks_to_fetch).await?;
 
     let mut local_receipts: Vec<views::ReceiptView> = vec![];
-    let mut outcomes = fetch_outcomes(&client, block.header.hash).await?;
+    let mut shards_outcomes = fetch_outcomes(&client, block.header.hash).await?;
     let mut indexer_chunks: Vec<IndexerChunkView> = vec![];
 
     for chunk in chunks {
         let views::ChunkView { transactions, author, header, receipts: chunk_non_local_receipts } =
             chunk;
-
-        let indexer_transactions = transactions
-            .into_iter()
-            .map(|transaction| {
-                IndexerTransactionWithOutcome {
-                    outcome: outcomes.remove(&transaction.hash).expect("The transaction execution outcome should always present in the same block as the transaction itself"),
-                    transaction,
-                }
+        let mut outcomes = shards_outcomes.get_mut(&header.shard_id).unwrap().clone();
```

Comment:
1. Let's always use `.expect` with a message why we believe this cannot fail.
2. Use [`.remove()`](https://doc.rust-lang.org/stable/std/collections/struct.HashMap.html#method.remove) instead of `get()` + `clone()`